internal/characters/faruzan/asc.go

```diff
@@ -0,0 +1,58 @@
+package faruzan
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+)
+
+const (
+	a4Key    = "faruzan-a4"
+	a4ICDKey = "faruzan-a4-icd"
+)
+
+// When characters affected by The Wind's Secret Ways' Prayerful Wind's Gift
+// deal Anemo DMG using Normal, Charged, Plunging Attacks, Elemental Skills, or
+// Elemental Bursts to opponents, they will gain the Hurricane Guard effect:
+// This DMG will be increased based on 32% of Faruzan's Base ATK. 1 instance of
+// Hurricane Guard can occur once every 0.8s. This DMG Bonus will be cleared
+// after Prayerful Wind's Benefit expires or after the effect is triggered
+// once.
+func (c *char) a4() {
+	c.Core.Events.Subscribe(event.OnEnemyHit, func(args ...interface{}) bool {
+		atk := args[1].(*combat.AttackEvent)
+		if atk.Info.Element != attributes.Anemo {
+			return false
+		}
+
+		switch atk.Info.AttackTag {
+		case combat.AttackTagNormal,
+			combat.AttackTagExtra,
+			combat.AttackTagPlunge,
+			combat.AttackTagElementalArt,
+			combat.AttackTagElementalArtHold,
+			combat.AttackTagElementalBurst:
+		default:
+			return false
+		}
+
+		active := c.Core.Player.ByIndex(atk.Info.ActorIndex)
+		if active.StatusIsActive(burstBuffKey) && !c.StatusIsActive(a4ICDKey) {
+			c.AddStatus(a4Key, 6, false)
```

Comment:
Do even need this a4Key? Nothing in the descsription indicates you need this. If you remove it you can then further reduce nesting by making the burst buff and icd check into early returns.